prime-router/metadata/hl7_mapping/ORU_R01/base/observation-result.yml

```
@@ -155,29 +155,29 @@ elements:
     hl7Spec: [ '%{hl7OBXField}-19' ]
 
   - name: performing-organization-name-pracrole
-    condition: '%observation.performer.resolve() is PractitionerRole'
-    resource: '%resource.performer.resolve().organization.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists()'
+    resource: '%resource.performer.resolve().ofType(PractitionerRole).organization.resolve()'
     schema: ../../common/datatype/xon-organization
     constants:
       hl7OrgField: '%{hl7OBXField}-23'
 
   - name: performing-organization-name-non-pracrole
-    condition: '%observation.performer.resolve().is(PractitionerRole).not()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists().not()'
```

When performer has both a `PractitionerRole` and an `Organization`, this seems like it would cause issues. I'm thinking what we really want is more like this
```suggestion
    condition: '%resource.exists() and %resource.is(PractitionerRole).not()'
```
or
```suggestion
    condition: '%resource.exists() and %resource.is(Organization)'
```

---

Yes, I think you are correct and the current fact that it was working relied on the ordering of the elements and not the conditions.  Will update

---

Did some testing on this in your branch today and got some clarity. OBX.16/OBX.23/OBX.24 work fine on HL7->FHIR->HL7 in the branch, but it's a bit messed up on HL7>FHIR. Basically prior to the change you commented on [here](https://github.com/CDCgov/prime-reportstream/pull/11174/files#r1332117929) (which isn't merged into your branch yet), OBX.23/24 were being added to the `PractitionerRole` for OBX.16 so we were populating all three fields from the same `performer`, even though they should be pulled from one `PractitionerRole` `performer` and one `Organization` `performer` respectively. This will basically need to be adjusted once the latest from master gets pulled in here. The good news is that the hci Translation Tests do cover this case, so you won't have to concoct a test specially for this.